src/logger/logmodule.hpp

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef LOGMODULE_HPP_
+#define LOGMODULE_HPP_
+
+#include <aos/common/tools/log.hpp>
+
+/**
+ * Log module type.
+ */
+class LogModuleType {
+public:
+    enum class Enum {
+        eApp,
+        eConfig,
+        eDatabase,
+        eVISIdentifier,
+        eIAMClient,
+        eIAMServer,
+        eUtils,
+        eNumModules,
+    };
+
+    static const aos::Array<const char* const> GetStrings()
+    {
+        static const char* const sLogModuleTypeStrings[] = {
+            "app",
+            "config",
+            "database",
+            "visidentifier",
+            "iamclient",
+            "iamserver",
+            "utils",
+        };
+
+        return aos::Array<const char* const>(sLogModuleTypeStrings, aos::ArraySize(sLogModuleTypeStrings));
+    };
+};
+
+using LogModuleEnum = LogModuleType::Enum;
+using LogModule     = aos::EnumStringer<LogModuleType>;
+
+/**
+ * Converts IAM LogModuleEnum to aos::LogModuleEnum.
+ *
+ * @param module IAM log module.
+ * @return aos::LogModuleEnum.
+ */
+constexpr auto AosLogModule(LogModuleEnum module)
+{
+    return static_cast<aos::LogModuleEnum>(
+        static_cast<int>(aos::LogModuleEnum::eNumModules) + static_cast<int>(module));
```

Comment:
It's not clear why eNumModules required for enum convertion.

Reply:
Because we do math operations on an enum which is declared as an enum class it's not possible without cast to do it